src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -283,6 +283,17 @@ public virtual void Write(object value)
             }
         }
 
+        /// <summary>
+        /// Equivalent to Write(stringBuilder.ToString()) however it uses the
+        /// StringBuilder.GetChunks() method to avoid creating the intermediate string 
+        /// </summary>
+        /// <param name="value">The string (as a StringBuilder) to write to the stream</param>
+        public virtual void Write(StringBuilder value)
```

Comment:
Should we be overriding these in any of the derived types? For example, seems like SyncTextWriter should override these to surround the call to the base in a lock, so the whole write operation is atomic and not just for the individual chunks.  And I expect IndentededTextWriter would need to override these as well to more appropriately deal with tabs, such that there's only one output at the beginning of the write rather than one for each chunk (though that's in corefx and would be updated after this coreclr change went in).  There are probably some optimizations that can be done as well, e.g. StringWriter could probably just copy the input StringBuilder to its internal one rather than going through the iterator, though that implementation is in corefx.

Reply:
I updated SyncTextWriter.   It is not clear to me why IndentTextWriter needs updating (Write(ReadOnlySpan) should already do the right thing), but that is a something that is part of CoreFX.

Reply:
> It is not clear to me why IndentTextWriter needs updating (Write(ReadOnlySpan) should already do the right thing), but that is a something that is part of CoreFX.

I think you're right.  I was looking at the implementation and saw that each call to Write was then calling OutputTabs, and I assumed that meant it would indent each call to Write, but now looking at OutputTabs, it conditionalizes its behavior on whether a new line was just written.

Reply:
Well, if the stream is a socket stream, there is more efficient way to write a sequence of buffers than writing them one by one, isn't there?

Reply:
> Well, if the stream is a socket stream, there is more efficient way to write a sequence of buffers than writing them one by one, isn't there?

Using StreamWriter?  It buffers.

Reply:
Ah my bad. I was thinking about raw Socket (and mathods that take collections of buffers), not NetworkStream.